src/tcgwars/logic/impl/gen4/MajesticDawn.groovy

```diff
@@ -2726,11 +2725,11 @@ public enum MajesticDawn implements LogicCardInfo {
               checkNoSPC()
               assert my.deck : "Your deck is empty"
               powerUsed()
-              def min = Math.min(2, my.deck.size())
-              def cards = my.deck.search(min:min, max:2, "Search your deck for any 2 cards", {true})
-              rearrange(cards)
+              def list = my.deck.select(count:2)
+              list = rearrange(list)
+              my.deck.removeAll(list)
```

Comment:
I 100% do not like using add/remove/addAll/removeAll on any persistent CardLists.

Reply:
Honestly went with what Mallow already did, since I recalled that one working fine ~~after quite a bit of changes, iirc~~. Do you happen to remember any alternative card that rearranges multiple cards in a different way, and which I could take as a base?

Reply:
Will mark this card as "Needs Implementation" in Trello, and look into it later (should work fine with `moveAll` in some way, just needs some testing while updating it).